Sources/UniRedis/UniRedis.swift

```diff
@@ -53,27 +54,31 @@ public class UniRedis {
 	}
 
 	public init(_ url: String) throws {
-		guard let match = try url.match("^(?:redis(\\+sentinel)?://)?(?:([^@]+)@)?([^:/]+)(?::([0-9]+))?(?:/([0-9]+))?$"), match.count == 5 else {
+		guard let match = try url.match("^(?:redis(\\+sentinel)?://)?(?:([^:@]+):)?(?:([^@]+)@)?([^:/]+)(?::([0-9]+))?(?:/([0-9]+))?$"), match.count == 6 else {
```

Comment:
This seems to break when redis port number is present, ex. redis://user:pass@127.0.0.1:6380/
Let's keep original regexp and split user:pass later, in the code below, see next comment. This way, we could also avoid renumbering match[] indexes.